error_test.go

```diff
@@ -244,7 +244,7 @@ func TestError(t *testing.T) {
 			}
 			assert.True(t, resp.IsError())
 			err := opensearch.ParseError(resp)
-			assert.Equal(t, err.Error(), http.StatusText(http.StatusUnauthorized))
+			assert.Contains(t, err.Error(), http.StatusText(http.StatusUnauthorized))
```

Comment:
Please adjust the tests to use Equal, so we expect a 1 to 1 match and not only part of the string.
```suggestion
			assert.Equal(t, err.Error(), fmt.Sprintf("invalid body: %q", http.StatusText(http.StatusUnauthorized))
```

Reply:
Checking for equality in error strings is actually an anti-pattern because they are not part of the API, as clearly shown by this failure. If we really want to check for errors, they should be done via errors.Is or errors.As. Do you still want me to change to `assert.Equal`?